titan/interactions/pca.py

```diff
@@ -26,27 +26,33 @@ def interact(model: "model.HIVModel", rel: "agent.Relationship") -> bool:
         """
         if not model.params.features.pca or model.time < model.params.pca.start_time:
             return False
-
         assert (
             model.params.model.network.enable
         ), "Network must be enabled for pca interactions"
 
         params = model.params.partnership.pca.frequency[rel.bond_type]
-        acts_prob = model.run_random.random()
-        acts_bin = 1
-        current_p_value = params[acts_bin].prob
-
-        while acts_prob > current_p_value:
-            acts_bin += 1
-            current_p_value += params[acts_bin].prob
-
-        min = params[acts_bin].min
-        max = params[acts_bin].max
 
-        if min == max:
-            num_acts = min
+        if params.type == "bins":
```

Comment:
since we also do this in other places, is it time to make it a utility?

Reply:
I think that's been an issue for a while....... Should definitely move that priority up!